site/content/en/docs/Installation/upgrading.md

```diff
@@ -0,0 +1,116 @@
+to the new version's cluster (green), and ensure nothing surprising happens during this process.
+
+This also allows easy rollback to the previous infrastrucutre that we already know to be working in production, with
+minimal interruptions to player experience.
+
+The following are steps to implement this:
+
+1. Create a new cluster of the same size or smaller as the current cluster.
+2. Install the new version of Agones on the new cluster.
+3. Deploy the same set of Fleets and/or GameServers from the old cluster into the new cluster.
+4. With your matchmaker, start sending a small percentage of your matched players' game sessions to the new cluster.
+5. Assuming everything is working successfully on the new cluster, slowly increase the percentage of matched sessions to the new cluster, until you reach 100%.
+6. Once you are comfortable with the stability of the new cluster with the new Agones version, shut down the old cluster.
+7. Congratulations - you have now upgraded to a new version of Agones! 👍
+
+### Single Cluster
+
+If you are upgrading a single cluster, we recommend creating a maintenance window, in which your game goes offline
+for the period of your upgrade, as there will be a short period in which Agones will be non-responsive during the upgrade.
+
+### Installation with install.yaml
+
+If you installed [Agones with install.yaml]({{< relref "_index.md#install-with-yaml" >}}), then you will need to delete
+the previous installation of Agones before upgrading to the new version, as we need to remove all of Agones before installing
+the new version.
+
+1. Start your maintenance window.
+1. Delete the current set of Fleets and/or GameServers in your cluster.
+1. Make sure to delete the same version of Agones that was previously installed, for example:
+   `kubectl delete -f https://raw.githubusercontent.com/googleforgames/agones/<old-release-version>/install/yaml/install.yaml`
+1. Install Agones [with install.yaml]({{< relref "_index.md#install-with-yaml" >}}).
+1. Deploy the same set of Fleets and/or GameServers back into the cluster.
+1. Run any other tests to ensure the Agones installation is working as expected.
+1. Close your maintenance window.
+7. Congratulations - you have now upgraded to a new version of Agones! 👍
+
+### Installation with Helm
+
+Helm features capabilities for upgrading to newer versions without having to delete the older version. For details on
+how to use Helm for upgrades, see the [helm upgrade](https://helm.sh/docs/helm/#helm-upgrade) documentation.
+
+Given the above, the steps for upgrade are simpler:
+
+1. Start your maintenance window.
+1. Run `helm upgrade` with the appropriate arguments, such a `--version`, for your specific upgrade
+1. Run any other tests to ensure the Agones installation is working as expected.
+1. Close your maintenance window.
+7. Congratulations - you have now upgraded to a new version of Agones! 👍
+
+## Upgrading Kubernetes
```

Comment:
So I'm not sure what the right text here is. I like what you have here re: risk on minor vs patch upgrades, but I wonder if the text about doing them "in place" is the best advice. My concern is that if you do it in place, you have to have a maintenance window, because you can't do with live gameservers running. Even for patch upgrades, doing it with two clusters means you can do it while your game is active, without downtime.

So I'm not sure what the right thing to say here is :man_shrugging: Maybe just the section on differences in risk between versions? Especially if you have custom code that also interacts with the K8s api?

Reply:
This one is tricky. I'm not sure if it's reasonable to force new cluster creation and workload migration each time there is a security patch (which seem relatively frequent). But I see your point about maintenance windows if you do in-place.... Leaving it as-is seems reasonable for now, and we can work on some of the trickier edge cases in a follow up.

Reply:
Agreed on tricky. Also, depending on your Kubernetes control layer (cloud, kubeadm, etc), there are likely different techniques you can deploy. e.g. nodepools are GKE specific, but maybe with a combo of regional cluster + multple nodepools, you can do a complete upgrade in cluster, within minimal/practically no downtime?